packages/material-ui-styles/src/makeStyles/makeStyles.d.ts

```diff
@@ -70,7 +72,7 @@ export type StylesHook<S extends Styles<any, any>> = StylesRequireProps<S> exten
   ? (props?: any) => ClassNameMap<ClassKeyOfStyles<S>>
   : (props: PropsOfStyles<S>) => ClassNameMap<ClassKeyOfStyles<S>>;
 
-export default function makeStyles<S extends Styles<any, any>>(
-  styles: S,
-  options?: WithStylesOptions<ThemeOfStyles<S>>,
-): StylesHook<S>;
+export default function makeStyles<C extends string, P extends {}, T = Theme>(
+  styles: StyleRules<P, C> | StyleRulesCallback<T, P, C>,
```

Comment:
Can we use the public type instead?
```suggestion
  styles: Styles<T, P, C>,
```